pkg/charts/charts.go

```diff
@@ -109,12 +109,9 @@ func (c *Chart) RenderNoPartials() (bytes.Buffer, error) {
 }
 
 // ReadFile updates the buffered file with the data read from disk
-func ReadFile(dir string, f *chartutil.BufferedFile) error {
+func ReadFile(fs http.FileSystem, dir string, f *chartutil.BufferedFile) error {
 	filename := dir + f.Name
-	if dir == "" {
-		filename = filename[7:]
-	}
-	file, err := static.Templates.Open(filename)
+	file, err := fs.Open(filename)
```

Comment:
Here you're reading directly from `fs`. When doing `linkerd install`, this ends up being `static.WithDefaultChart(helmDefaultChartDir)` which points directly to the "real" file system.
Before, we had `static.Templates.Open(filename)` which hits the "real" file system when `!prod` and the VFS when `prod`. So this is why it's failing the integration test (trying to hit the real FS instead of the VFS).

Reply:
Yep, That makes sense, As `static.Templates` is replaced by the logic to read from a VFS with code generation which our helper code is not doing similarly.